android/app/src/main/java/app/organicmaps/widget/SpeedLimitView.java

```diff
@@ -0,0 +1,225 @@
+package app.organicmaps.widget;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Rect;
+import android.graphics.Typeface;
+import android.util.AttributeSet;
+import android.util.Pair;
+import android.view.View;
+
+import androidx.annotation.ColorInt;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import app.organicmaps.R;
+import app.organicmaps.util.StringUtils;
+
+public class SpeedLimitView extends View
+{
+  private interface DefaultValues
+  {
+    @ColorInt
+    int BACKGROUND_COLOR = Color.WHITE;
+    @ColorInt
+    int BORDER_COLOR = Color.RED;
+    @ColorInt
+    int ALERT_COLOR = Color.RED;
+    @ColorInt
+    int TEXT_COLOR = Color.BLACK;
+    @ColorInt
+    int TEXT_ALERT_COLOR = Color.WHITE;
+
+    float BORDER_WIDTH_RATIO = 0.1f;
+  }
+
+  @ColorInt
+  private final int mBackgroundColor;
+
+  @ColorInt
+  private final int mBorderColor;
+
+  @ColorInt
+  private final int mAlertColor;
+
+  @ColorInt
+  private final int mTextColor;
+
+  @ColorInt
+  private final int mTextAlertColor;
+
+  @NonNull
+  private final Paint mSignBackgroundPaint;
+  @NonNull
+  private final Paint mSignBorderPaint;
+  @NonNull
+  private final Paint mTextPaint;
+
+  private float mWidth;
+  private float mHeight;
+  private float mBackgroundRadius;
+  private float mBorderRadius;
+  private float mBorderWidth;
+
+  private double mSpeedLimit;
+  @Nullable
+  private String mSpeedLimitStr;
```

Comment:
I understand that you want the data in these variables to be in different formats. That's fine, but it’s difficult to understand without knowing the details of the implementation. So, please either rename them (e.g., `mSpeedLimitMps`, `mSpeedLimitPlatformUnits`) or add a comment with this information.